b9lab-content/3-main-concepts/15-migrations.md

```diff
@@ -1,5 +1,75 @@
 # Migrations
 
+## On-chain Upgrades
+
+A Cosmos SDK application running on a Cosmos blockchain can be upgraded in an orderly, on-chain fashion. 
+
+Upgrading blockchains and blockchain applications is notoriously difficult and risky. Cosmos SDK solves for the common risks and challenges. What are those challenges and risks that Cosmos SDK solves?
+
+Generally, when a blockchain is upgraded it is vital that all nodes upgrade simultaneously and at the same block height. In a disorderly setting, this is difficult to actually achieve. If the nodes do not do so then the blockchain will "fork" into two blockchains with common history - one chain that observes the new rules and one chain that observes the old rules. It is generally not possible for the two chains to reach a common consensus or merge together in the future. 
+
+<HighlightBox type="info">
+Without software support for upgrades, upgrading a live chain is risky because all of the validators need to pause their state machines at exactly the same block height and apply the upgrade before resuming. If this is not done correctly, there can be state inconsistencies which are hard to recover from.
+</HighlightBox>
+
+Smart contracts on EVM chains such as Ethereum are immutible software. By definition, they are difficult or impossible to change. Various strategies based on modularity can simulate the effects of upgrading the contracts but all known methods have inherent limitations. Chief among the limitations are the difficulties, impossibility or prohibitive cost of reorganizing data at rest. This places a significant limitation on the types of upgrades that are feasible. 
+
+A Cosmos SDK blockchain built for a specific application can be upgraded without forks and, if necessary, the existing data can be reorganized to prepare it for use by a new version of the application and blockchain. 
+
+## Process Overview
+
+### Plan
+
+A "Plan" is an upgrade process to take place at a specific block height in the future. It includes a SideCar that executes when the upgrade process commences, a name of the Plan and block height at which to execut. Importantly, acceptance or rejection of the Plan is managed through the normal governance process. A "cancel proposal" can be submitted and adopted which prevents the accepted Plan from executing. Cancellation is contingent on knowing that a given Plan is a poor idea before the upgrade happens.  
+
+The "Info" in a Plan kicks off the Sidecar process.
+
+```shell
+type Plan struct {
+  Name   string
+  Height int64
+  Info   string
+}
+```
+
+### Sidecar Process
+
+A "Sidecar" is a binary the nodes can run to attend to processes outside of Cosmos binaries. This can include steps such as downloading and compiling software from a certain commit in a repo.
```

Comment:
this is not the first time "sidecar" appears, please add explanation of "sidecar" the first time it is used in this doc, or refer to here.